ambry-replication/src/main/java/com.github.ambry.replication/ReplicationMetrics.java

```diff
@@ -368,6 +369,27 @@ public Long getValue() {
     replicaLagInBytes.add(replicaLag);
   }
 
+  /**
+   * Tracks the number of partitions for which replication is disabled.
+   * @param replicaThreadPools A map of datacenter names to {@link ReplicaThread}s handling replication from that
+   *                           datacenter
+   */
+  public void trackReplicationDisabledPartitions(final Map<String, List<ReplicaThread>> replicaThreadPools) {
+    for (Map.Entry<String, List<ReplicaThread>> entry : replicaThreadPools.entrySet()) {
+      String datacenter = entry.getKey();
+      List<ReplicaThread> pool = entry.getValue();
+      Gauge<Integer> disabledCount = () -> {
+        Set<PartitionId> replicationDisabledPartitions = new HashSet<>();
+        for (ReplicaThread replicaThread : pool) {
+          replicationDisabledPartitions.addAll(replicaThread.getReplicationDisabledPartitions());
+        }
+        return replicationDisabledPartitions.size();
+      };
+      registry.register(MetricRegistry.name(ReplicaThread.class, "ReplicationDisabledPartitions-" + datacenter),
+          disabledCount);
+    }
+  }
```

Comment:
do you think if we have another global metric which tracks the total no of disabled partitions (across DCs) ?